### PR TITLE
Add upgade instructions for v0.14

### LINKED ISSUE #142: Document v0.13 to v0.14 upgrade
We should document the v0.13 to v0.14 upgrade.

Things to be noted

- Introduction of the legacy manifests (which to use now)
- Webhook now being required
- Deployment selector issue

### PR DESCRIPTION
**Pull request by meyskens**
Fixes #142

### CODE REVIEW

**Comment by meyskens** [comment]
needs release-0.14 branch
/hold

File: content/en/docs/installation/upgrading/upgrading-0.13-0.14.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+This step will not affect any existing certificates but will stop renewal or new issuance while upgrading.
+
+
+Version `v0.14` now comes in 2 versions of static manifests, you will need to use the correct new one:
+* Kubernetes 1.15 or higher: you can use the normal `cert-manager.yaml`
+* Kubernetes 1.14 or lower: you have to now use the `cert-manager-legacy.yaml` version
+* OpenShift 4: you can now the normal `cert-manager.yaml`
+* OpenShift 3: you have to now use the `cert-manager-legacy.yaml` version instead of the OpenShift version
+
+> **Note**: If you're using the `cert-manager-legacy.yaml` version you not have API version conversion and thus only support `cert-manager.io/v1apha2` API resources.
```
Comment (munnerz) [inline content/en/docs/installation/upgrading/upgrading-0.13-0.14.md]: “you not have”

File: content/en/docs/installation/upgrading/upgrading-0.13-0.14.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+If you used the `no-webhook` version you will now have to install a version with the webhook.
```
Comment (munnerz) [inline content/en/docs/installation/upgrading/upgrading-0.13-0.14.md]: Maybe rephrase to “The webhook is now a required component, meaning that `no-webhook` variant of the manifests are no longer available in this release. Please use the appropriate manifests as mentioned above according to your Kubernetes version.”? It currently sort of reads like they need to **specifically** choose a version with the webhook enabled (ie implies there is a version without still)

File: content/en/docs/installation/upgrading/upgrading-0.13-0.14.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+The webhook is now a required component of cert-manager.
+
+From here on you can find Follow the [regular upgrade process(../).
```
Comment (munnerz) [inline content/en/docs/installation/upgrading/upgrading-0.13-0.14.md]: “you can find Follow the regular upgrade process” reads a bit weirdly

Comment (meyskens) [reply content/en/docs/installation/upgrading/upgrading-0.13-0.14.md]: oops...

File: content/en/docs/installation/upgrading/upgrading-0.13-0.14.md (hunk 2 above)
Comment (meyskens) [reply content/en/docs/installation/upgrading/upgrading-0.13-0.14.md]: Goon one!

File: content/en/docs/installation/upgrading/upgrading-0.13-0.14.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+* Kubernetes 1.15 or higher: you can use the normal `cert-manager.yaml`
+* Kubernetes 1.14 or lower: you have to now use the `cert-manager-legacy.yaml` version
+* OpenShift 4: you can now the normal `cert-manager.yaml`
+* OpenShift 3: you have to now use the `cert-manager-legacy.yaml` version instead of the OpenShift version
+
+> **Note**: If you're using the `cert-manager-legacy.yaml` version you will not have API version conversion and thus only support `cert-manager.io/v1apha2` API resources.
+
+The webhook is now a required component, meaning that `no-webhook` variant of the manifests are no longer available in this release. Please use the appropriate manifests as mentioned above according to your Kubernetes version.
+
+From here on you can follow the [regular upgrade process(../).
```
Comment (munnerz) [inline content/en/docs/installation/upgrading/upgrading-0.13-0.14.md]: Missing closing `]`

File: content/en/docs/installation/upgrading/upgrading-0.13-0.14.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+```
+
+This will delete the deployment so they can be replaced when you apply the upgrade.
+This step will not affect any existing certificates but will stop renewal or new issuance while upgrading.
+
+
+Version `v0.14` now comes in 2 versions of static manifests, you will need to use the correct new one:
+* Kubernetes 1.15 or higher: you can use the normal `cert-manager.yaml`
+* Kubernetes 1.14 or lower: you have to now use the `cert-manager-legacy.yaml` version
+* OpenShift 4: you can now the normal `cert-manager.yaml`
```
Comment (munnerz) [inline content/en/docs/installation/upgrading/upgrading-0.13-0.14.md]: `you can now the normal`

File: content/en/docs/installation/upgrading/upgrading-0.13-0.14.md [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+before upgrading:
+
+```bash
+$ kubectl delete -n cert-manager deployment cert-manager cert-manager-cainjector cert-manager-webhook
+```
+
+This will delete the deployment so they can be replaced when you apply the upgrade.
+This step will not affect any existing certificates but will stop renewal or new issuance while upgrading.
+
+
```
Comment (munnerz) [inline content/en/docs/installation/upgrading/upgrading-0.13-0.14.md]: nit: extra newline here - is that correct?

File: content/en/docs/installation/upgrading/upgrading-0.13-0.14.md [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+title: "Upgrading from v0.13 to v0.14"
+linkTitle: "v0.13 to v0.14"
+weight: 30
+type: "docs"
+---
+
+Due to changes in the Deployment selector you will need to remove the deployments first before being able to upgrade.
+
+If you are using a deployment tool that automatically handles this (i.e. Helm),
+there should be no additional action to take.
```
Comment (munnerz) [inline content/en/docs/installation/upgrading/upgrading-0.13-0.14.md]: Is this true?

Comment (munnerz) [reply content/en/docs/installation/upgrading/upgrading-0.13-0.14.md]: (that Helm handles it automatically)

Comment (meyskens) [reply content/en/docs/installation/upgrading/upgrading-0.13-0.14.md]: Gonna test this in alpha 1 now!

Comment (meyskens) [reply content/en/docs/installation/upgrading/upgrading-0.13-0.14.md]: Okay... I wasn't able to fully test it with alpha 0, alpha 1 now clearly fails... we have to add instructions for that

Comment (meyskens) [reply content/en/docs/installation/upgrading/upgrading-0.13-0.14.md]: Updated it, tested it on helm 2 and 3 and they seem fine with deleting the deployments

File: content/en/docs/installation/upgrading/upgrading-0.13-0.14.md [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+This will delete the deployment so they can be replaced when you apply the upgrade.
+This step will not affect any existing certificates but will stop renewal or new issuance while upgrading.
+
+Version `v0.14` now comes in 2 versions of static manifests, you will need to use the correct new one:
+* Kubernetes 1.15 or higher: you can use the normal `cert-manager.yaml`
+* Kubernetes 1.14 or lower: you have to now use the `cert-manager-legacy.yaml` version
+* OpenShift 4: you can now use the normal `cert-manager.yaml`
+* OpenShift 3: you have to now use the `cert-manager-legacy.yaml` version instead of the OpenShift version
+
+> **Note**: If you're using the `cert-manager-legacy.yaml` version you will not have API version conversion and thus only suppor2t `cert-manager.io/v1apha2` API resources.
```
Comment (munnerz) [inline content/en/docs/installation/upgrading/upgrading-0.13-0.14.md]: `suppor2t` 🙈

Comment (meyskens) [reply content/en/docs/installation/upgrading/upgrading-0.13-0.14.md]: fixed
